source/Ship.cpp

```diff
@@ -2407,11 +2407,21 @@ double Ship::DisabledHull() const
 
 
 
-int Ship::JumpsRemaining() const
+int Ship::JumpsRemaining(bool followParent) const
 {
 	// Make sure this ship has some sort of hyperdrive, and if so return how
 	// many jumps it can make.
-	double jumpFuel = JumpFuel(targetSystem);
+	double jumpFuel = 0.;
+	if(!targetSystem && followParent)
+	{
+		// If this ship has no destination, the parent's substitutes for it,
+		// but only if the location is reachable.
+		auto p = GetParent();
+		if(p)
+			jumpFuel = JumpFuel(p->GetTargetSystem());
```

Comment:
GetTargetSystem() will return nullptr if the parent arrives at it's destination.  It would seem that you would use parent's current system in that case?

Reply:
I'm not sure that escorts should go to the current parent's system and if it has meaningfulness in what kind of condition.